src/app/modules/main/chat_section/model.nim

```diff
@@ -431,19 +401,25 @@ QtObject:
     defer: modelIndex.delete
     self.dataChanged(modelIndex, modelIndex, @[ModelRole.Muted.int])
 
-  proc changeCanPostValues*(self: Model, id: string, canPostReactions, viewersCanPostReactions: bool) =
+  proc changeCanPostValues*(self: Model, id: string, canPost, canView, canPostReactions, viewersCanPostReactions: bool) =
     let index = self.getItemIdxById(id)
     if index == -1:
       return
-    if(self.items[index].canPostReactions == canPostReactions and
+    if(self.items[index].canView == canView and
+        self.items[index].canPost == canPost and
+        self.items[index].canPostReactions == canPostReactions and
         self.items[index].viewersCanPostReactions == viewersCanPostReactions
       ):
       return
+    self.items[index].canPost = canPost
+    self.items[index].canView = canView
     self.items[index].canPostReactions = canPostReactions
     self.items[index].viewersCanPostReactions = viewersCanPostReactions
     let modelIndex = self.createIndex(index, 0, nil)
     defer: modelIndex.delete
     self.dataChanged(modelIndex, modelIndex, @[
+      ModelRole.CanPost.int,
+      ModelRole.CanView.int,
       ModelRole.CanPostReactions.int,
       ModelRole.ViewersCanPostReactions.int
     ])
```

Comment:
Would be nice to make the comparison one by one and emit the actual list of changed roles. 
Definitely a minor thing, but now when it's 4 properties it makes more sense to do I guess

Reply:
Fixed